step.yml

```diff
@@ -182,14 +182,21 @@ inputs:
       title: "App URL"
       description: |-
         Required when you have selected _App file (URL)_ for _App type_.
+        Note that _Bundle ID_ is also required when you select _iOS_ for _OS_ and _Remote TestKit_ or _Remote TestKit Onpremise_ for _Environment_ due to their restriction.
       is_expand: true
   - bundle_id: 
     opts:
       title: "Bundle ID"
       description : |-
-        Required when you have selected _iOS_ for _OS_ and _Installed_ for _App type_.
-
+        Specify the unique ID for the iOS app under test. 
         * ex) `com.apple.Preferences`
+
+        This field is required when you select _iOS_ for _OS_ and one of the following conditions.
+        1. When you select _Installed_ for _App type_.
+        2. When you select _Remote TestKit_ for _Environment_ and _App file (URL)_ for _App type_
+        3. When you select _Remote TestKit_ for _Environment_ and _App file (cloud upload)_ for _App type_
+        4. When you select _Remote TestKit Onpremise_ for _Environment_ and _App file (URL)_ for _App type_
+        5. When you select _Remote TestKit Onpremise_ for _Environment_ and _App file (cloud upload)_ for _App type_
```

Comment:
How about following conditions?

1. When you select _Installed_ for _App type_.
2. When you select _Remote TestKit_ or _Remote TestKit Onpremise_ for _Environment_ (required for any _App type_)

Reply:
If 1 happens, you need `Bundle ID` even what an _Environment_ comes. 
And if 2 happens, you also need `Bundle ID` even what an _App type_ comes.

Did I answer your question? But I'm feeling like it looks somewhat redundant...

Reply:
> If 1 happens, you need `Bundle ID` even what an _Environment_ comes.
> And if 2 happens, you also need `Bundle ID` even what an _App type_ comes.

If so, I think you can simply write down the two points in the description. Of course the initial ones are not bad, but very long...